### DISCUSSION TITLE
help me with right setting on rmq

### DISCUSSION THREAD

**Discussion by bookvik** [mikub/titanoboa#38] [Q&A] [answered]
titanoboa_1    | ERROR [main] - Error starting system :core-rmq !
titanoboa_1    | clojure.lang.ExceptionInfo: Error in component :mq-connection in system com.stuartsierra.component.SystemMap calling #'com.stuartsierra.component/start {:reason :com.stuartsierra.component/component-function-threw-exception, :function #'com.stuartsierra.component/start, :system-key :mq-connection, :component #titanoboa.channel.rmq.RMQConnectionComponent{:params {:host "localhost", :port 5672, :username "guest", :password "guest", :vhost "/", :connection-name "titanoboa-connection"}, :connection nil}, :system #<SystemMap>}

![image](https://user-images.githubusercontent.com/204606/113298202-3e576800-9304-11eb-9e13-59c15013bc47.png)
![image](https://user-images.githubusercontent.com/204606/113298401-76f74180-9304-11eb-8c4c-8b8428b13fb5.png)

I tried to insert the host in different places and systems config and in front of it as it is now in the screenshot.

But I always always get something that the code does not apply my value

although I see that this parameter is present

![image](https://user-images.githubusercontent.com/204606/113298598-ad34c100-9304-11eb-8d69-51821726749b.png)

I will attach the dependency file at the same time

**Comment by bookvik** [comment]
just assembled a new build where I replaced the host with mine, apparently somewhere there is not enough merge

I will find it if I correct it

**Reply by mikub** [reply]
@bookvik  What version are you running? I added the ability to configure rmq host only recently in 1.0.0-alfa , so either it is broken or you might have an older version... 
Once I have some time I will post a sample system config for it as well.

**Reply by mikub** [reply]
@bookvik here it is:
https://github.com/mikub/titanoboa/blob/cluster/src/clj/titanoboa/system/rabbitmq.clj#L21

You can see that under `:systems-config` -> `core-rmq` you might need to add following keys:
```clojure
:mq-host 
:mq-port 
:mq-username 
:mq-password 
:mq-vhost
```

**Reply by bookvik** [reply]
i am using the latest version : 1 alpha

I have not yet managed to send the completed tasks on the rabbit to the postgres archive

**Reply by mikub** [reply]
@bookvik see a sample config for RMQ here:
https://github.com/mikub/titanoboa/wiki/Clustering#server-config-file
i.e. you have to add ` :archival-queue` and make sure it is the same for the archival system:

```clojure
:systems-catalogue {:core  {:system-def #'titanoboa.system.rabbitmq/distributed-core-system
                                                         :worker-def #'titanoboa.system.rabbitmq/distributed-worker-system
                                                         :autostart  true
                                                         :worker-count 2
                                                         :restart-workers-on-error true}
                                                 :archival-system {:system-def #'titanoboa.system.rabbitmq/archival-system
                                                                   :autostart  true}}
```

```clojure
:systems-config {:core {:new-jobs-queue "titanoboa-bus-queue-0"
                                                     :jobs-queue "titanoboa-bus-queue-1"
                                                     :archival-queue "titanoboa-bus-archive"
                                                     :eviction-age (* 1000 60 5)
                                                     :eviction-interval (* 1000 30)}
                                              :archival-system {:jdbc-url "jdbc:postgresql://localhost:5432/mydb?currentSchema=titanoboa"
                                                                :archival-queue "titanoboa-bus-archive"}}
```

**Reply by bookvik** [reply]
thank you. I'll try to do it today

**Reply by bookvik** [reply]
awesome! it is worked perfectly

**Comment by mikub** [comment, marked as answer]
Answered in these two comments:

https://github.com/mikub/titanoboa/discussions/38#discussioncomment-558516

https://github.com/mikub/titanoboa/discussions/38#discussioncomment-564469
